File: src/006/p006.scala
object Main {
  def square(n: Int) = n * n

  def solve(): Int = {
    val numbers = 1 to 100
    square(numbers.sum) - numbers.map(square).sum
  }

  def main(args: Array[String]): Unit = {
    val result = solve()
    println(result)
  }
}

File: src/006/p006.clj
; works in both clojure and clojurescript

(defn solve []
  (let [numbers (range 1 101)
        square #(* % %)
        [sum-of-squares sum] (transduce (map (juxt square identity)) (partial mapv +) [0 0] numbers)]
        (- (square sum) sum-of-squares)))

(prn (solve))